lec04-dplyr.Rmd

```diff
@@ -1,4 +1,724 @@
+
+```{r}
+surveys %>% 
+    summarize_all(n_distinct)
+```
+
+Remember that there are still `NA` values here, that's why there appears to be three sexes although there is only male and female. There are four taxa so that could be a good candidate, let's see which those are.
+
+```{r}
+surveys %>% 
+    distinct(taxa)
+```
+
+It seems reasonable that these taxa contain animals different enough to have diverse weights and length of their feet. Lets use this categorical variable to color the scatter plot. 
+
+```{r}
+ggplot(surveys, aes(x = weight, y = hindfoot_length, color = taxa)) +
+    geom_point(alpha = 0.2)
+```
+
+Only rodents? That was unexpected... Let's check what's going on.
+
+```{r}
+surveys %>% 
+    group_by(taxa) %>% 
+    tally()
+```
+
+There is definitely mostly rodents in our data set...
+
+```{r}
+surveys %>% 
+    filter(!is.na(hindfoot_length)) %>% # control by removing `!`
+    group_by(taxa) %>% 
+    tally()
+```
+
+...and it turns out that only rodents, have had their hindfeet measured!
+
+Let's remove all animals that did not have their hindfeet measured, including
+those rodents that did not. Animals without their weight measured will also be removed.
+
+```{r}
+surveys_hf_wt <- surveys %>% 
+    filter(!is.na(hindfoot_length) & !is.na(weight))
+
+surveys_hf_wt %>% 
+    summarize_all(n_distinct)
+```
+
+Maybe the genus can explain what we are seeing.
+
+```{r}
+ggplot(surveys_hf_wt, aes(x = weight, y = hindfoot_length, color = genus)) +
+    geom_point(alpha = 0.2)
+```
+
+Now this looks good! There is a clear separation between different genus, but also significant spread within genus, for example in the weight of the green Neotoma observations. There are also two clearly separate clusters that are both colored in olive green (Dipodomys). Maybe separating the observations into different speciese would be better?
+
+```{r}
+ggplot(surveys_hf_wt, aes(x = weight, y = hindfoot_length, color = species)) +
+    geom_point(alpha = 0.2)
+```
+
+Great! Together with the genus plot, this definitely seem to explain most of the
+variance we see in the hindfoot length and weight measurements. It is still a
+bit messy as it appears like we have around 5 clusters, but there are 21 species
+in the legend.
+
+```{r}
+surveys %>% 
+    filter(!is.na(hindfoot_length) & !is.na(weight)) %>% 
+    group_by(species) %>% 
+    tally() %>% 
+    arrange(desc(n))
+```
+
+There is a big drop from 838 to 159, let's include only those with more than 800 observations.
+
+```{r}
+abundant_species <- surveys %>% 
+    filter(!is.na(hindfoot_length) & !is.na(weight)) %>% 
+    group_by(species) %>% 
+    tally() %>% 
+    arrange(desc(n)) %>% 
+    filter(n > 800) %>% # 
+    select(species)
+
+abundant_species
+```
+
+```{r}
+surveys_abun_species <- surveys %>% 
+    filter(!is.na(hindfoot_length) &
+        !is.na(weight) &
+        species %in% abundant_species$species)
```

Comment:
nice and clever use of `%in%` with species column!

Reply:
Thanks! I considered creating `abundant_species` as a vector directly (with `.$` or `pull()`), but figured the current approach would be more intuitive and not introduce new functions for them to remember.